Makefile

```diff
@@ -1,15 +1,41 @@
+#!/usr/bin/env bash
+
 # Application
-APP_ROOT := $(CURDIR)
+APP_ROOT := $(abspath $(lastword $(MAKEFILE_LIST))/..)
 APP_NAME := twitcher
 
-# Anaconda
-CONDA := $(shell command -v conda 2> /dev/null)
-ANACONDA_HOME := $(shell conda info --base 2> /dev/null)
-CONDA_ENV := $(APP_NAME)
+# Conda
+CHECK      ?= shell command -v
+CONDA_HOME ?= $($(CHECK) conda info --base 2> /dev/null)
+CONDA_ENV  ?= $(APP_NAME)
+CONDA_BIN  := $(CONDA_HOME)/bin/conda
+CONDA_TARGET_PREFIX := readlink -e "$(CONDA_HOME)/envs/$(CONDA_ENV)" 2> /dev/null
+CONDA_ACTUAL_PREFIX := readlink -e "${CONDA_PREFIX}" 2> /dev/null
```

Comment:
`CONDA_PREFIX` is not always defined. On my mac I'm using anaconda system installation and I have to run `conda activate` to make conda available.

Reply:
The idea here is that, I `CONDA_PREFIX` is indeed not defined, therefore the conda environment is not activated, so the script activates it with `CONDA_CMD` before doing any other operation (pip install, etc.).
If the "right" `CONDA_ENV == twitcher` conda environment is already activated because the user already did `conda activate twitcher`, `CONDA_PREFIX` points to it and there is no need for `CONDA_CMD` to be executed.